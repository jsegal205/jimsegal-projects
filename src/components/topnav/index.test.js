import React from "react";
import { BrowserRouter as Router } from "react-router-dom";

import { render, cleanup } from "@testing-library/react";
import TopNav from "./index";

afterEach(cleanup);

describe("TopNav Component", () => {
  it("renders correctly", () => {
    const { container } = render(
      <Router>
        <TopNav />
      </Router>
    );
    expect(container).toMatchSnapshot();
  });
});
