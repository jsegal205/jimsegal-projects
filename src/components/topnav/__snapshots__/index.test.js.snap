// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopNav Component Does not render on root route 1`] = `<div />`;

exports[`TopNav Component Renders on all other routes 1`] = `
<div>
  <nav>
    <ul
      class="topnav-list"
    >
      <li
        class="topnav-list-item"
      >
        <a
          href="https://jimsegal.com"
          title="OG site with other projects"
        >
          JimSegal.com
        </a>
      </li>
      <li
        class="topnav-list-item"
      >
        <a
          href="/congress"
          title="Some fun data surrounding the current congressional sesssion"
        >
          Congress
        </a>
      </li>
      <li
        class="topnav-list-item"
      >
        <a
          href="/isJimWearingShorts"
          title="A site to predict if Jim is wearing shorts or not"
        >
          Is Jim Wearing Shorts?
        </a>
      </li>
      <li
        class="topnav-list-item"
      >
        <a
          href="mountaingoat"
          title="Digital game"
        >
          Mountain Goat Game
        </a>
      </li>
      <li
        class="topnav-list-item"
      >
        <a
          href="/recipes"
          title="A bunch of recipes that I frequent and wanted to show off"
        >
          Recipes
        </a>
      </li>
      <li
        class="topnav-list-item"
      >
        <a
          href="/spacex"
          title="A countdown timer until the next launch of a Space X rocket"
        >
          Space X
        </a>
      </li>
    </ul>
  </nav>
  <hr />
</div>
`;
